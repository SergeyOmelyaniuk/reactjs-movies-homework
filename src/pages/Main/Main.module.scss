.main {
	padding: 30px 0px 60px 0px;
	@media (max-width: 1600px) {
		padding: 25px 0px 50px 0px;
	}
}

.wrapCategoriesSort {
	display: flex;
	justify-content: flex-end;
	margin-bottom: 30px;
	@media (max-width: 1600px) {
		margin-bottom: 25px;
	}
}

.wrapPagination {
	display: flex;
	justify-content: center;
	margin-top: 60px;
	@media (max-width: 1600px) {
		margin-top: 50px;
	}
}
