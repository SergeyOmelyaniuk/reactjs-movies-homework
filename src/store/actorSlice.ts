import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
import { Actor, MovieAPI, MovieImages } from '../types';

export const fetchActorProfile = createAsyncThunk(
	'actor/fetchActorProfile',
	async function (id: string, { rejectWithValue, getState }) {
		const state = getState() as any;
		const { languageSelected } = state.language;

		try {
			const [actor, images, movies] = await Promise.all([
				fetch(
					`https://api.themoviedb.org/3/person/${id}?api_key=${process.env.REACT_APP_API_KEY}&language=${languageSelected.value}`
				),
				fetch(
					`https://api.themoviedb.org/3/person/${id}/images?api_key=${process.env.REACT_APP_API_KEY}`
				),
				fetch(
					`https://api.themoviedb.org/3/person/${id}/combined_credits?api_key=${process.env.REACT_APP_API_KEY}&language=${languageSelected.value}`
				),
			]).then((data) => Promise.all(data.map((res) => res.json())));

			return { actor, images, movies };
		} catch (error) {
			return rejectWithValue(error.message);
		}
	}
);

export interface ActorState {
	status: string | null;
	actorProfile: Actor | null;
	actorImages: MovieImages[] | null;
	actorKnowBy: MovieAPI[] | null;
}

const initialState: ActorState = {
	status: null,
	actorProfile: null,
	actorImages: null,
	actorKnowBy: null,
};

const actorReducer = createSlice({
	name: 'actor',
	initialState,
	reducers: {},
	extraReducers: (builder) => {
		builder.addCase(fetchActorProfile.pending, (state) => {
			state.status = 'loading';
		});
		builder.addCase(
			fetchActorProfile.fulfilled,
			//TODO
			(state, { payload }: PayloadAction<any>) => {
				state.status = 'fulfilled';
				state.actorProfile = payload.actor;
				state.actorImages = payload.images.profiles;
				state.actorKnowBy = payload.movies.cast;
			}
		);
		builder.addCase(fetchActorProfile.rejected, (state) => {
			state.status = 'failed';
		});
	},
});

export default actorReducer.reducer;
