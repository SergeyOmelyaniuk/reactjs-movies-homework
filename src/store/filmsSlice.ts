import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import { translate } from '../constants';
import getGenresFilms from '../helpers/getGenresFilms';

export const fetchFilms = createAsyncThunk(
	'todos/fetchTodos',
	async function (_, { rejectWithValue, getState, dispatch }) {
		const { films } = getState() as any;

		if (films.listGenres.length === 0) {
			await dispatch(fetchGenres());
		}

		const state = getState() as any;
		const { currentPage, categoryValue, listGenres } = state.films;

		try {
			const response = await fetch(
				`https://api.themoviedb.org/3/movie/${categoryValue}?api_key=75375a34318f83bb07dacbf870f89dd3&language=en-US&page=${currentPage}`
			);

			if (!response.ok) {
				throw new Error('Server Error!');
			}

			const data = await response.json();
			console.log(data);

			const dataWithGenres = getGenresFilms(listGenres, data.results);

			return dataWithGenres;
		} catch (error) {
			return rejectWithValue(error.message);
		}
	}
);

export const fetchGenres = createAsyncThunk(
	'todos/fetchGenres',
	async function (_, { rejectWithValue }) {
		try {
			const response = await fetch(
				`https://api.themoviedb.org/3/genre/movie/list?api_key=75375a34318f83bb07dacbf870f89dd3&language=en-US`
			);

			if (!response.ok) {
				throw new Error('Server Error!');
			}

			const data = await response.json();

			return data;
		} catch (error) {
			return rejectWithValue(error.message);
		}
	}
);

//TODO
export interface FilmsState {
	categoryValue: string;
	pages: number;
	currentPage: number;
	listfilms: any;
	status: any;
	error: any;
	listGenres: any;
}

const initialState: FilmsState = {
	categoryValue: translate.en.categories[0].value,
	pages: 6,
	currentPage: 1,
	listfilms: [],
	status: null,
	error: null,
	listGenres: [],
};

const filmsReducer = createSlice({
	name: 'films',
	initialState,
	reducers: {
		changeCategory(state, action) {
			state.categoryValue = action.payload;
		},
		setCurrentPage(state, action) {
			state.currentPage = action.payload;
		},
	},
	extraReducers: (builder) => {
		builder.addCase(fetchFilms.pending, (state, { payload }) => {
			state.status = 'loading';
			state.error = null;
		});
		builder.addCase(fetchFilms.fulfilled, (state, { payload }) => {
			state.listfilms = payload;
		});
		builder.addCase(fetchFilms.rejected, (state, { payload }) => {
			state.status = 'failed';
			state.error = payload;
		});

		builder.addCase(fetchGenres.fulfilled, (state, { payload }) => {
			state.listGenres = payload.genres;
		});
	},
});

export const { changeCategory, setCurrentPage } = filmsReducer.actions;

export default filmsReducer.reducer;
