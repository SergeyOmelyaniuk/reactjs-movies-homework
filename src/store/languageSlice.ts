import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { languages } from '../constants';
import { Language } from '../types';

export interface LanguageState {
	languageSelected: Language;
}

const initialState: LanguageState = {
	languageSelected: languages[0],
};

const languageReducer = createSlice({
	name: 'language',
	initialState,
	reducers: {
		setCurrentlanguage(state, action: PayloadAction<Language>) {
			state.languageSelected = action.payload;
		},
	},
});

export const { setCurrentlanguage } = languageReducer.actions;

export default languageReducer.reducer;
