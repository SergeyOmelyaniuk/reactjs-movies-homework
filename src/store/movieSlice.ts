import { createSlice, createAsyncThunk, PayloadAction } from '@reduxjs/toolkit';
import { MovieAPI, Genre, MovieDetails, MovieImages, Actor } from '../types';

export const fetchMovieDetails = createAsyncThunk(
	'movie/fetchMovieDetails',
	async function (id: string, { rejectWithValue, getState }) {
		const state = getState() as any;
		const { languageSelected } = state.language;
		// const { currentIdFilm } = state.film;
		//TODO

		try {
			const [movie, images, recommendations, actors] = await Promise.all([
				fetch(
					`https://api.themoviedb.org/3/movie/${id}?api_key=${process.env.REACT_APP_API_KEY}&language=${languageSelected.value}`
				),
				fetch(
					`https://api.themoviedb.org/3/movie/${id}/images?api_key=${process.env.REACT_APP_API_KEY}`
				),
				fetch(
					`https://api.themoviedb.org/3/movie/${id}/recommendations?api_key=${process.env.REACT_APP_API_KEY}&language=${languageSelected.value}&page=1`
				),
				fetch(
					`https://api.themoviedb.org/3/movie/${id}/credits?api_key=${process.env.REACT_APP_API_KEY}&language=${languageSelected.value}`
				),
			]).then((data) => Promise.all(data.map((res) => res.json())));

			return { movie, images, recommendations, actors };
		} catch (error) {
			return rejectWithValue(error.message);
		}
	}
);

export interface MovieState {
	movieDetails: MovieDetails | null;
	movieImages: MovieImages[] | null;
	moviesRecommendations: MovieAPI[] | null;
	topCast: Actor[] | null;
	status: string | null;
	listGenres: Genre[];
	//TODO
	// currentIdFilm: number;
}

const initialState: MovieState = {
	movieDetails: null,
	movieImages: null,
	moviesRecommendations: null,
	topCast: null,
	status: null,
	listGenres: [],
	//TODO
	// currentIdFilm: 460465,
};

const movieReducer = createSlice({
	name: 'movies',
	initialState,
	reducers: {},
	extraReducers: (builder) => {
		builder.addCase(fetchMovieDetails.pending, (state) => {
			state.status = 'loading';
		});
		builder.addCase(
			fetchMovieDetails.fulfilled,
			//TODO
			(state, { payload }: PayloadAction<any>) => {
				state.movieDetails = payload.movie;
				state.movieImages = payload.images.backdrops;
				state.moviesRecommendations = payload.recommendations.results;
				state.topCast = payload.actors.cast;
				state.status = 'fulfilled';
			}
		);
		builder.addCase(fetchMovieDetails.rejected, (state) => {
			state.status = 'failed';
		});
	},
});
//TODO
// export const { } = movieReducer.actions;

export default movieReducer.reducer;
