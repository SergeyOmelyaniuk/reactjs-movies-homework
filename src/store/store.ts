import { configureStore } from '@reduxjs/toolkit';
import languageReducer from './languageSlice';
import moviesReducer from './moviesSlice';
import movieReducer from './movieSlice';

export const store = configureStore({
	reducer: {
		language: languageReducer,
		movies: moviesReducer,
		movie: movieReducer,
	},
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
