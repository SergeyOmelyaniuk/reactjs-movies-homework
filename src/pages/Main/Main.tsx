import React, { useEffect } from 'react';
import CategoriesSort from '../../сomponents/CategoriesSort';
import Pagination from '../../сomponents/Pagination';
import WrapMovies from '../../сomponents/WrapMovies';
import styles from './Main.module.scss';
import { translate } from '../../constants';

import { useSelector } from 'react-redux';
import { RootState } from '../../store/store';

import { useDispatch } from 'react-redux';
import {
	changeCategory,
	setCurrentPage,
	fetchFilms,
} from '../../store/filmsSlice';

function Main() {
	const categoryValue = useSelector(
		(state: RootState) => state.films.categoryValue
	);
	const pages = useSelector((state: RootState) => state.films.pages);
	const currentPage = useSelector(
		(state: RootState) => state.films.currentPage
	);
	const listFilms = useSelector((state: RootState) => state.films.listfilms);

	const dispatch = useDispatch();

	const hundlerChangeCategory = (value: string) => {
		dispatch(changeCategory(value));
	};
	const hundlerSetCurrentPage = (value: number) => {
		dispatch(setCurrentPage(value));
	};

	useEffect(() => {
		dispatch(fetchFilms());
	}, [dispatch, currentPage, categoryValue]);

	return (
		<div className={styles.main}>
			<div className={styles.wrapCategoriesSort}>
				<CategoriesSort
					categories={translate.en.categories}
					categoryValue={categoryValue}
					changeCategory={hundlerChangeCategory}
				/>
			</div>
			<WrapMovies movies={listFilms} />
			<div className={styles.wrapPagination}>
				<Pagination
					pages={pages}
					setCurrentPage={hundlerSetCurrentPage}
					currentPage={currentPage}
				/>
			</div>
		</div>
	);
}

export default Main;
