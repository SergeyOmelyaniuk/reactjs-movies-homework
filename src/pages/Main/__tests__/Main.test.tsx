import React from 'react';
import { render } from '@testing-library/react';
import Main from '..';
import { Provider } from 'react-redux';
import { store } from '../../../store/store';
import { setSearcValue } from '../../../store/filmsSlice';

describe('Main', () => {
	test('should render with the list of movies', () => {
		const { container } = render(
			<Provider store={store}>
				<Main />
			</Provider>
		);

		expect(container).toMatchSnapshot();
	});
	test('should render with search results', () => {
		const { container } = render(
			<Provider store={store}>
				<Main />
			</Provider>
		);

		store.dispatch(setSearcValue('some search text'));

		expect(container).toMatchSnapshot();
	});
});
