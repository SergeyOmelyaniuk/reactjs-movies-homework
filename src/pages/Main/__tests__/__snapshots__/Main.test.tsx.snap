// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main should render 1`] = `
<div>
  <div
    class="main"
  >
    <div
      class="wrapCategoriesSort"
    >
      <div
        class="categoriesSort"
      >
        <input
          checked=""
          id="radio-1"
          name="sort"
          type="radio"
          value="popular"
        />
        <label
          for="radio-1"
        >
          Popular
        </label>
        <input
          id="radio-2"
          name="sort"
          type="radio"
          value="top_rated"
        />
        <label
          for="radio-2"
        >
          Top rated
        </label>
        <input
          id="radio-3"
          name="sort"
          type="radio"
          value="upcoming"
        />
        <label
          for="radio-3"
        >
          Upcoming
        </label>
      </div>
    </div>
    <div
      class="wrapMovies"
    >
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              68
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Красотка на взводе
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
              <span>
                Комедия
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              72
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Сокровище Счастливчика
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
              <span>
                Комедия
              </span>
              <span>
                Драма
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              76
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Сталинград
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
              <span>
                Комедия
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              69
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Люди в черном: Интернэшнл
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
              <span>
                Комедия
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              96
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Значит, война
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              77
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Приключения желтого чемоданчика
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              79
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Поезд в Пусан
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              79
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Никто не выжил
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              53
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Ведьма
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              53
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Звёздные войны: Пробуждение силы
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              58
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Марсианин
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              100
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Республика Z
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              90
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Планета 51
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              86
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Красные туфельки и семь гномов
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              92
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Капитошка
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              58
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Война будущего
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              86
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Чернобыль
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              81
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Странный дом
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              79
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Дальний космос
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
      <div
        class="wrapMovie"
      >
        <div
          class="movieCard"
        >
          <div
            class="wrapImage"
          >
            <div
              class="overlay"
            />
            <img
              alt=""
              class="image"
            />
            <span
              class="rating"
            >
              20
            </span>
            <div
              class="preview"
            />
          </div>
          <div
            class="content"
          >
            <h2
              class="title"
            >
              Мандалорец
            </h2>
            <p
              class="genre"
            >
              <span>
                Боевик
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="wrapPagination"
    >
      <ul
        class="pagination"
      >
        <li
          class="item active"
        >
          1
        </li>
        <li
          class="item"
        >
          2
        </li>
        <li
          class="item"
        >
          3
        </li>
        <li
          class="item"
        >
          4
        </li>
        <li
          class="item"
        >
          5
        </li>
      </ul>
    </div>
  </div>
</div>
`;
